### PR TITLE
Add remaining init tests

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

File: jujugui/static/gui/src/app/test-init.js [hunk 1]
```diff
@@ -54,17 +55,582 @@ describe('init', () => {
...
+
+    it('can start and stop the drag timer: _dragLeaveTimerControl', () => {
+      app._dragleaveTimerControl('start');
+      assert.equal(app._dragLeaveTimer !== undefined, true);
+      app._dragleaveTimerControl('stop');
+      assert.equal(app._dragLeaveTimer === null, true);
+    });
+  });
+
+  xdescribe('Application authentication', () => {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/test-init.js]: There are a couple of tests in here that are failing. @hatched if you had a minute to have a look that'd be great.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8120/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8125/

File: jujugui/static/gui/src/app/test-init.js (hunk 1 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/test-init.js]: I've figured these out now.

File: jujugui/static/gui/src/app/init.js [hunk 2]
```diff
@@ -772,15 +777,6 @@ class GUIApp {
       }));
   }
 
-  /**
-    Hide the drag notifications.
-  */
-  _hideDragOverNotification() {
```
Comment (hatched) [inline jujugui/static/gui/src/app/init.js]: oops, thanks

File: jujugui/static/gui/src/app/init.js [hunk 3]
```diff
@@ -35,6 +35,11 @@ const yui = window.yui;
 class GUIApp {
   constructor(config) {
     /**
+      Stash the location object so that tests can override it.
+      @type {Object}
+    */
+    this.location = window.location;
```
Comment (hatched) [inline jujugui/static/gui/src/app/init.js]: Is this necessary? Do we actually use this in the app anywhere or was it only added to appease a test? I'm pretty sure that is an artifact of the old Y.App routing.

Comment (huwshimi) [reply jujugui/static/gui/src/app/init.js]: You're right, nothing was failing before I added that, except for the test that checks it exists.

File: karma.conf.js [hunk 4]
```diff
@@ -77,7 +78,7 @@ module.exports = function(config) {
     preprocessors: {
       'jujugui/static/gui/src/app/components/**/test-*.js': ['browserify'],
       'jujugui/static/gui/src/app/test-*.js': ['browserify'],
-      'jujugui/static/gui/src/app/init/test-*.js': ['browserify'],
+      'jujugui/static/gui/src/app/init/*.js': ['browserify'],
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply karma.conf.js]: Not sure why but it couldn't find some require files until I did this, but I just tried reverting it and it's all working again.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8128/

**Pull request merged**